src/main/java/de/craftlancer/clstuff/citizensets/CitizenSetFunction.java

```diff
@@ -0,0 +1,235 @@
+package de.craftlancer.clstuff.citizensets;
+
+import de.craftlancer.core.util.ParticleUtil;
+import de.craftlancer.core.util.Tuple;
+import org.bukkit.Color;
+import org.bukkit.Particle;
+import org.bukkit.configuration.serialization.ConfigurationSerializable;
+import org.bukkit.entity.Player;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public abstract class CitizenSetFunction implements ConfigurationSerializable {
+    private String id;
+    
+    CitizenSetFunction(String id) {
+        this.id = id;
+    }
+    
+    public CitizenSetFunction(Map<String, Object> map) {
+        this.id = (String) map.get("id");
+    }
+    
+    public abstract Consumer<Tuple<Player, Long>> getConsumer();
```

Comment:
Is this function ever called from outside this class? If not it doesn't need to be exposed.